numba/cpython/unicode.py

```diff
@@ -831,51 +832,53 @@ def _adjust_indices(length, start, end):
 
 
 @overload_method(types.UnicodeType, 'startswith')
-def unicode_startswith(s, substr, start=None, end=None):
-    if not (start is None or isinstance(start, (types.Omitted,
-                                                types.Integer,
-                                                types.NoneType))):
-        raise TypingError('When specified, the arg "start" must be an Integer')
+def unicode_startswith(s, prefix, start=None, end=None):
+    if not is_nonelike(start) and not isinstance(start, types.Integer):
+        raise TypingError(
+            "When specified, the arg 'start' must be an Integer or nonelike")
 
-    if not (end is None or isinstance(end, (types.Omitted,
-                                            types.Integer,
-                                            types.NoneType))):
-        raise TypingError('When specified, the arg "end" must be an Integer')
+    if not is_nonelike(end) and not isinstance(end, types.Integer):
+        raise TypingError(
+            "When specified, the arg 'end' must be an Integer or nonelike")
 
-    if isinstance(substr, (types.Tuple, types.UniTuple)):
-        def startswith_tuple_impl(s, substr, start=None, end=None):
-            for item in substr:
+    if isinstance(prefix, (types.Tuple, types.UniTuple)):
+        if isinstance(prefix, types.Tuple) or \
+                not isinstance(prefix.dtype, types.UnicodeType):
+            raise TypingError(
+                "The arg 'prefix' should a UniTuple of UnicodeType")
```

Comment:
Just accept `types.UniTuple` with a `.dtype` which is `types.UnicodeType` ? e.g.

```python
if isinstance(prefix, types.UniTuple) and isinstance(prefix.dtype, types.UnicodeType):
    ... # then return the implementation
```